src/app/router/ui/ProtectedRoute.tsx

```diff
@@ -12,7 +12,7 @@ export const ProtectedRoute = () => {
 
     useEffect(() => {
         if (!account) {
-            navigate(RoutesPaths[Routes.MAIN]);
+            navigate(RoutesPaths[Routes.LOGIN], { state: { from: window.location.pathname } });
```

Comment:
what if user opens protected route initially 
eg open browser and put .../protectedRoute Based on current logic system redirects user back to browser main page after successfully login 
Think how you can handle it here or in onSuccess callback for signIn